gcapi/gcapi.py

```diff
@@ -433,3 +440,69 @@ def __call__(
             return response.json()
         else:
             return response
+
+    def run_external_algorithm(self, algorithm_name, file_to_upload):
+        """
+            This function uploads an input image to grand challenge and runs an already uploaded algorithm which the
+            user has access to. If the upload is finished correctly, grand challenge will automatically submit a job
+            which you can access via AlgorithmJobsAPI. After the job is finished successfully the result will be
+            available via AlgorithmResultsAPI.
+
+            Parameters
+            ----------
+            algorithm_name: str
+                Title of the algorithm which has already been uploaded.
+            file_to_upload: os.path
+                Full path of the image to be uploaded
+
+            Returns
+            -------
+            upload_session_pk: UUID
+            This can be used to construct a query like /api/v1/cases/images/?origin__pk=upload_session_pk to find out
+            which Image did this RawImageUploadSession give rise to. This can be further used to identify the submitted
+            job.
+            """
+        _, filename = os.path.split(file_to_upload)
+        algorithm = list(
+            filter(
+                lambda a: a["title"] == algorithm_name,
+                self.algorithms.list()["results"],
+            )
+        )
+        if not algorithm:
+            raise IOError(
+                "{} is not found in available list of algorithms".format(algorithm_name)
+            )
+        algorithm_image = algorithm[0]["algorithm_container_images"][0]
+        self.chunked_uploads.send(file_to_upload)
+        uploaded_file_list = list(
+            filter(
+                lambda a: a["filename"] == filename,
+                self.chunked_uploads.list()["results"],
+            )
+        )
+        if not uploaded_file_list:
+            raise IOError("{} has not been uploaded properly.".format(filename))
+        staged_file_id = uploaded_file_list[0]["uuid"]
+        raw_image_upload_session_create_data = {"algorithm_image": algorithm_image}
+        raw_image_upload_session_create_response = self.raw_image_upload_sessions.create(
+            **raw_image_upload_session_create_data
+        )
+        upload_session_pk = raw_image_upload_session_create_response["pk"]
+        raw_image_upload_session_files_create_data = {
+            "upload_session": raw_image_upload_session_create_response["api_url"],
+            "staged_file_id": staged_file_id,
+            "filename": filename,
+            "validate": False,
```

Comment:
My comment earlier was about all of these `validate: False` statements. Why is this one here?

Reply:
I never received any response from you regarding whether you would like to verify against a schema. validate is True by default except delete request.

Reply:
Where did you ask that? You marked the conversation as resolved with no comment. What is the reason for skipping validation here?

Reply:
I think on your slack. For this one, I was just checking whether removing validate works or not.

Reply:
"Hi James, in gcapi APIBase does not have any create/partial_update methods but ModifiableMixin does. Should I inherit from ModifiableMixin on addition to APIBase or add create/partial_update to APIBase. Also, is there a need to validate on the client side like Harm does?"
January 16th

Reply:
Slack is not a reliable way to get answers - I get hundreds of messages per day there and they're likely to get lost (especially after 3 weeks). Address issues about code in GitHub.

Reply:
I see. Sorry.